### PR TITLE
Add the first implementation of line filtering

### PR DESCRIPTION
**Pull request by llucax**
This is a custom superfence for pymdown-extensions that can filter lines by specifying a set of line ranges to show.

The ranges are specified in the `show_lines` option, which is a comma-separated list of ranges, where each range is in the format `start:end` or `line`, where `start`, `end` and `line` are positive integers.

If `start` is empty, it is assumed to be 1. If `end` is empty, it is assumed to be the end of the file. If only `line` is used, it is assumed to be both `start` and `end` (so allowing just one line).

Lines are 1-indexed.

Warnings are emitted if the `show_lines` option is invalid. There is a hacky way to determine if we are running inside MkDocs or not, and if we are then we use the MkDocs logger, so warnings are shown in the MkDocs output with special formatting and also are detected as such when running in *strict* mode. If we are not running inside MkDocs, then we use the logger for this module.

### CODE REVIEW

**Comment by llucax** [comment]
Rendered documentation can be found here: https://llucax.github.io/frequenz-pymdown-superfence-filter-lines-python

File: src/frequenz/pymdownx/superfences/filter_lines/__init__.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+        Args:
+            text: String to parse.
+
+        Returns:
+            The created ranges, or `None` if no ranges are given, plus a list of errors,
+                if any.
+        """
+        ranges: set[LinesRange] = set()
+        errors: list[ValueError] = []
+        for n, r in enumerate(text.split(","), start=1):
```
Comment (Marenz) [inline src/frequenz/pymdownx/superfences/filter_lines/__init__.py]: maybe use `range` or `range_str` here instead of `r`?

Comment (llucax) [reply src/frequenz/pymdownx/superfences/filter_lines/__init__.py]: Updated

File: tests/test_do_format.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    md = mock.MagicMock()
+    preprocessor = mock.MagicMock()
+    preprocessor.highlight.side_effect = True
+    md.preprocessors.return_value = {"fenced_code_block": preprocessor}
+    yield md
+
+
+_SOURCE = """\
+1. This is some text
+2. which have multiple lines
```
Comment (Marenz) [inline tests/test_do_format.py:32]: *has   XD

**Review by Marenz — APPROVED** [review]

File: tests/test_do_format.py (hunk 2 above)
Comment (llucax) [reply tests/test_do_format.py:32]: hehe, will fix in a followup PR

**Pull request merged**